presto-raptor/src/main/java/com/facebook/presto/raptor/storage/OrcStorageManager.java

```diff
@@ -256,6 +264,20 @@ public ConnectorPageSource getPageSource(
         }
     }
 
+    private static int addSpecialIndex(long columnId)
+    {
+        if (isShardRowIdColumn(columnId)) {
+            return ROWID_COLUMN;
+        }
+        if (isShardUuidColumn(columnId)) {
+            return SHARD_UUID_COLUMN;
+        }
+        if (isBucketNumberColumn(columnId)) {
+            return BUCKET_NUMBER_COLUMN;
+        }
+        throw new IllegalArgumentException(format("Invalid column id %s", columnId));
```

Comment:
We shouldn't be throwing `INTERNAL_ERROR`. 
I can create a new error code for this case.